extra logging plus bypass on hanging bug gcp

bperian opened the pull request:
- added more debug logging lines 
- Explicit settings for Google Service regarding resumable uploads 
- Refactor read/write operations on blobs to bypass hanging process bug inside Google Insert request
<!--
Thank you for your interest in and contributing to Elasticsearch! There
are a few simple things to check before submitting your pull request
that can help with the review process. You should delete these items
from your submission, but they are here to help bring them to your
attention.
-->

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

Commit:
extra logging plus bypass on hanging bug gcp

clarkperkins commented on plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java:
```diff
@@ -259,19 +226,27 @@ void writeBlob(String blobName, InputStream inputStream, long blobSize) throws I
                     
                         logger.warn("[repository-gcs][writeBlob] Lost - Throttling try #" + retry );
-                        throw e;
+                        logger.warn(e);
```
I think we may need to throw an exception on the last retry - otherwise nobody will know if the blob fails to get written

clarkperkins commented on plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java:
```diff
@@ -259,19 +226,27 @@ void writeBlob(String blobName, InputStream inputStream, long blobSize) throws I
                     
                         logger.warn("[repository-gcs][writeBlob] Lost - Throttling try #" + retry );
-                        throw e;
+                        logger.warn(e);
```
I think we may need to throw an exception on the last retry - otherwise nobody will know if the blob fails to get written

bperian replied:
will add a commit today

clarkperkins commented on plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java:
```diff
@@ -197,55 +197,22 @@ boolean blobExists(String blobName) throws IOException {
                 
-                TimeUnit.SECONDS.sleep(2);
-                return readBlob(blobName,1);
+                logger.warn("[ReadBlob()] Lost - Throttling try #" + retry );
```
Same thing here - we should throw something on the last retry so that the user knows that something actually failed

bperian replied:
will add a commit today

Commit:
Enforce SLA

Theoretically the maximum failure is two times. This will enforce the SLA and drop the operation in case requests fail more than two times.
